pac4j-core/src/main/java/org/pac4j/core/authorization/OrMultiAuthorizer.java

```diff
@@ -0,0 +1,49 @@
+/*
+  Copyright 2012 - 2015 pac4j organization
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package org.pac4j.core.authorization;
+
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.profile.CommonProfile;
+
+/**
+ * Multi authorizers check based on OR logic.
+ *
+ * @param <U> the user profile
+ * @author Jerome Leleu
+ * @since 1.8.0
+ */
+public class OrMultiAuthorizer<U extends CommonProfile> implements Authorizer<U> {
+
+    protected final Authorizer<U>[] authorizers;
+
+    public OrMultiAuthorizer(final Authorizer<U>[] authorizers) {
+        this.authorizers = authorizers;
```

Comment:
You're right: that may be a problem. Though, if you take a look at the `HasAccessRoleAuthorizer`, the `Authorizer` classes are built internally. Maybe I should make this constructor `protected`. Will keep that somewhere in my mind ;-)